File: 14-Facade/src/go/amplifier.go
package main

import "fmt"

// Subsystem type — the audio amplifier.
type Amplifier struct{}

func (a Amplifier) On()                { fmt.Println("Amplifier on") }
func (a Amplifier) Off()               { fmt.Println("Amplifier off") }
func (a Amplifier) SetVolume(level int) { fmt.Printf("Amplifier volume set to %d\n", level) }

File: 14-Facade/src/rust/amplifier.rs
// Subsystem — the audio amplifier.
pub struct Amplifier;

impl Amplifier {
    pub fn on(&self) { println!("Amplifier on"); }
    pub fn off(&self) { println!("Amplifier off"); }
    pub fn set_volume(&self, level: i32) { println!("Amplifier volume set to {}", level); }
}
